app/controllers/api/v1/messages_controller.rb

```diff
@@ -91,14 +91,31 @@ def apply_filters(messages, options)
         %i[ids offer_id order_id item_id package_id].map do |f|
           messages = messages.send("filter_by_#{f}", options[f]) if options[f].present?
         end
-        messages = messages.with_state_for_user(current_user, options[:state].split(',')) if options[:state].present?
+
+        if options[:state].present? && %w[unread read].include?(options[:state])
+          messages = messages.with_state_for_user(current_user, options[:state].split(','))
+        end
+
+        if bool_param(:only_notification, false)
+          notification_ids = messages
+            .select("max(messages.id) AS message_id")
+            .group("messageable_type, messageable_id, is_private")
+          messages = messages.where("messages.id IN (?)", notification_ids).order("messages.id DESC")
+        end
+
         messages
       end
 
       def apply_scope(records, scope)
-        return records unless ALLOWED_SCOPES.include? scope
```

Comment:
A small trick that might be helpful here is to convert scope to an array regardless of it's type, and then you don't need to code separate `if is_a?(Array)` cases.

e.g.
```
scope = [scope].flatten.compact.uniq
> [nil] becomes []
> 1 becomes [1]
> [1,2] becomes [1, 2]
```